src/AppBundle/EventListener/OccurrenceListener.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/*
+ * This file is part of Eventbase API.
+ *
+ * (c) 2017–2018 ITK Development
+ *
+ * This source file is subject to the MIT license.
+ */
+
+namespace AppBundle\EventListener;
+
+use AppBundle\Entity\DailyOccurrence;
+use AppBundle\Entity\Occurrence;
+use AppBundle\Service\OccurrenceSplitterService;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Event\OnFlushEventArgs;
+use Doctrine\ORM\Event\PreFlushEventArgs;
+
+class OccurrenceListener
+{
+    private $occurrenceSplitter;
+
+    /**
+     * OccurrenceListener constructor.
+     *
+     * @param OccurrenceSplitterService $occurrenceSplitter
+     */
+    public function __construct(OccurrenceSplitterService $occurrenceSplitter)
+    {
+        $this->occurrenceSplitter = $occurrenceSplitter;
+    }
+
+    /**
+     * @param PreFlushEventArgs $args
+     *
+     * @throws \Exception
+     */
+    public function onFlush(OnFlushEventArgs $args): void
+    {
+        $em = $args->getEntityManager();
+
+        $this->insertDailyOccurrences($em);
+        $this->deleteDailyOccurrences($em);
+        $this->synchronizeDailyOccurrences($em);
+    }
+
+    /**
+     * Insert new DailyOccurrences matching the Occurrences scheduled for insertion
+     * in doctrines unit of work.
+     *
+     * @param EntityManager $em
+     *
+     * @throws \Doctrine\ORM\ORMException
+     */
+    private function insertDailyOccurrences(EntityManager $em): void
+    {
+        $uow = $em->getUnitOfWork();
+        $entities = $uow->getScheduledEntityInsertions();
+
+        $occurrences = array_filter($entities, function ($entity) {
+            return $entity instanceof Occurrence;
+        });
+
+        $classMetadata = $em->getClassMetadata(DailyOccurrence::class);
+
+        foreach ($occurrences as $occurrence) {
+            $dailyOccurrences = $this->occurrenceSplitter->getDailyOccurrences($occurrence);
+
+            foreach ($dailyOccurrences as $dailyOccurrence) {
+                $em->persist($dailyOccurrence);
+                $uow->computeChangeSet($classMetadata, $dailyOccurrence);
+            }
+        }
+    }
+
+    /**
+     * Synchronize the DailyOccurrences to match the Occurrences scheduled for updates
+     * in doctrines unit of work.
+     *
+     * @param EntityManager $em
+     *
+     * @throws \Doctrine\ORM\ORMException
+     */
+    private function synchronizeDailyOccurrences(EntityManager $em): void
```

Comment:
Should this be called `updateDailyOccurrences` (to match `insert…` and `delete…`)?

Reply:
`synchronize` is a deliberate choice because this function both updates, inserts and deletes.